punishments.js

```diff
@@ -216,7 +242,7 @@ Punishments.loadRoomPunishments = async function () {
 Punishments.savePunishments = function () {
 	FS(PUNISHMENT_FILE).writeUpdate(() => {
 		const saveTable = new Map();
-		Punishments.ips.forEach((punishment, ip) => {
+		Punishments.ips.forEach((/** @type {Punishment} */ punishment, /** @type {string} */ ip) => {
```

Comment:
You can stop typing this now that you've defined it in the `forEach` definition.

Reply:
er even with the forEach call back definition it's still complaining if I don't provide the annotations 
![image](https://user-images.githubusercontent.com/22896712/34392071-309a091e-eb07-11e7-94e7-b45c6fa83948.png)

Reply:
I think this has to do with the issue you suppressed as `possible TypeScript bug`. I'll investigate later I guess.